fabric-model-loading-api-v1/src/client/java/net/fabricmc/fabric/api/client/model/loading/v1/ModelLoadingPlugin.java

```diff
@@ -64,5 +64,20 @@ interface Context {
 		 * Event access to register model resource providers.
 		 */
 		Event<ModelResourceProvider> resourceProviders();
+
+		/**
+		 * Event access to monitor unbaked model loads and replace the loaded model.
+		 */
+		Event<UnbakedModelObserver> onUnbakedModelLoad();
+
+		/**
+		 * Event access to swap the unbaked model used for baking without replacing the loaded model.
+		 */
+		Event<UnbakedModelObserver> onUnbakedModelPreBake();
```

Comment:
The difference between this and the `load` one should be explained clearly in the javadoc. (i.e. replacing an unbaked model directly might be problematic if other models expect a specific class. For example, replacing a model used as the `parent` of a vanilla JSON model by a different class than `JsonUnbakedModel` will cause a failure).

Reply:
Added an explanation based on what we discussed on Discord.